main.tf

```diff
@@ -190,8 +190,8 @@ resource "aws_s3_bucket" "sftp_transfer" {
   bucket = var.bucket_name
 
   logging {
-    target_bucket = aws_s3_bucket.sftp_transfer_s3_logging[0].id
-    target_prefix = "/"
+    target_bucket = length(aws_s3_bucket.sftp_transfer_s3_logging) > 0 ? aws_s3_bucket.sftp_transfer_s3_logging[0].id : ""
```

Comment:
Pretty sure this resource will fail to apply in the false (no logging) case when target_bucket is set to an empty string.

Reply:
BTW, the case where I hit problems with _not_ having this logic was on a `destroy`.

Reply:
Setting to null would be enough to correct this if these config params were at the top level of the resource but since they are nested looks like the "[dynamic block](https://github.com/terraform-providers/terraform-provider-aws/issues/1110#issuecomment-514498331)" syntax is the solution (have never used this before)

Reply:
Per slack conversation, while the logging is not optional, managing the bucket is optional.  So I reworked this.